CalibCalorimetry/HcalAlgos/src/HcalSiPMnonlinearity.cc

```diff
@@ -0,0 +1,19 @@
+#include <iostream>
+#include "Math/Polynomial.h"
+#include "CalibCalorimetry/HcalAlgos/interface/HcalSiPMnonlinearity.h"
+
+// Assume parameters come to us from the reco side; i.e.,
+// true pes = corfun(pixelsfired). But we want to invert that.
+//
+int HcalSiPMnonlinearity::getPixelsFired(int inpes) const
+{
+  ROOT::Math::Polynomial p(a2,b1,c0,-inpes);
+  std::vector<double> roots = p.FindRealRoots();
```

Comment:
calling gsl directly would likely be better in the long term (seems this function is not yet used)